### PR TITLE
perf(ipam): allocation-free IP comparison in range engine

### PR DESCRIPTION
**Pull request by oilbeater**
## Summary
- Replace the `big.Int`-based `IP.LessThan`/`GreaterThan` (two heap `big.Int` allocations per call) with an allocation-free per-family compare. These are the innermost primitive of the range engine (`IPRangeList.Find` binary search, `Separate`/`Merge`/`Intersect`), exercised on every pod IP allocate/release.
- IPv4 compares as its 4-byte value (`To4` does not allocate for 4-byte storage), otherwise the 16-byte form is compared. Benchmark: **0 allocs/op, ~7 ns/op** vs the previous big.Int path.
- Fixes a latent inconsistency with `Equal`: the old raw-`SetBytes` compare mis-ordered mixed 4-byte/16-byte IPv4 representations and treated `0.0.0.0` as equal to `::`. Added unit tests for both edge cases.

## Test plan
- [x] `go test ./pkg/ipam/...` passes (incl. new `TestIPCompareMixedRepresentation`, `TestIPCompareV4ZeroVsV6Zero`)
- [x] `make lint` reports 0 issues
- [x] Benchmark confirms 0 allocs/op for `LessThan`
- [x] Existing `TestIPLessThan`/`TestIPGreaterThan` and full ipam suite unchanged behavior for in-practice (family-segregated) cases

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/ipam/ip.go [hunk 1]
```diff
@@ -40,12 +41,25 @@ func (a IP) Equal(b IP) bool {
...
+// that IPv4 addresses compare as their 4-byte value regardless of whether they
+// are stored in 4-byte or 16-byte (v4-in-v6) representation. This keeps the
+// result consistent with Equal and avoids the big.Int allocations that the
+// range engine would otherwise pay on every comparison.
+func (a IP) cmp(b IP) int {
+	if a4, b4 := net.IP(a).To4(), net.IP(b).To4(); a4 != nil && b4 != nil {
+		return bytes.Compare(a4, b4)
+	}
+	return bytes.Compare(net.IP(a).To16(), net.IP(b).To16())
+}
```
[the remark this answers is not in the document]
Comment (oilbeater) [reply pkg/ipam/ip.go:57]: Good catch on the precision. You're right that `To16()` on a 4-byte IPv4 allocates in the mixed-family branch. I've left the implementation as-is intentionally rather than adding a stack-buffer v4-in-v6 mapping: the range engine only ever compares same-family IPs (IP range lists are segregated into V4/V6), so the mixed-family branch is never reached on the hot path — the only thing exercising it is the determinism unit test. Both same-family paths (`To4` for IPv4, `To16` on already-16-byte IPv6) are allocation-free, which is the goal here. I've updated the doc comment to state this scope precisely instead of claiming unconditional allocation-freedom.
